eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/RawExceptionThrowCheck.java

```diff
@@ -0,0 +1,162 @@
+            insideServiceInterface = false;
+        }
+    }
+
+    @Override
+    public void visitToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.METHOD_DEF:
+                insideGeneratedMethod = hasGeneratedAnnotation(token);
+                break;
+            case TokenTypes.INTERFACE_DEF:
+                insideServiceInterface = hasServiceInterfaceAnnotation(token);
+                break;
+            case TokenTypes.LITERAL_THROW:
+                if (insideGeneratedMethod || insideServiceInterface) {
+                    return;
+                }
+                DetailAST expr = token.findFirstToken(TokenTypes.EXPR);
+                if (expr == null) {
+                    return;
+                }
+
+                DetailAST newToken = expr.findFirstToken(TokenTypes.LITERAL_NEW);
+                if (newToken != null) {
+                    String name = FullIdent.createFullIdentBelow(newToken).getText();
+                    if (IGNORED_EXCEPTIONS.stream().noneMatch(name::endsWith)) {
+                        log(newToken, ERROR_MESSAGE);
+                    }
+
+                    break;
+                }
+
+                DetailAST methodCallToken = expr.findFirstToken(TokenTypes.METHOD_CALL);
+                if (methodCallToken != null) {
+                    if (CORE_EXCEPTION_FACTORY_NAME.equals(FullIdent.createFullIdentBelow(methodCallToken).getText()))
+                    {
+                        log(methodCallToken, ERROR_MESSAGE);
+                    }
+                }
+                break;
+            case TokenTypes.METHOD_CALL:
+                String name = FullIdent.createFullIdentBelow(token).getText();
+                if (name.endsWith(".fromJson")) {
+                    insideGeneratedMethod = true;
+                }
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    private boolean hasGeneratedAnnotation(DetailAST methodDefAst) {
+        DetailAST modifiers = methodDefAst.findFirstToken(TokenTypes.MODIFIERS);
+        DetailAST annotation = modifiers.findFirstToken(TokenTypes.ANNOTATION);
+        if (annotation == null) {
+            return false;
+        }
+
+        String annotationName = annotation.findFirstToken(TokenTypes.IDENT).getText();
+        if (!annotationName.endsWith("Metadata")) {
+            return false;
+        }
+
+        DetailAST memberValuePairs = annotation.findFirstToken(TokenTypes.ANNOTATION_MEMBER_VALUE_PAIR);
+        if (memberValuePairs == null) {
+            return false;
+        }
+
+        DetailAST props = findFirstTokenByPredicate(memberValuePairs, p -> p.getType() == TokenTypes.IDENT && "properties".equals(p.getText())).orElse(null);
+        if (props == null) {
+            return false;
+        }
+
+        DetailAST argValue = memberValuePairs.findFirstToken(TokenTypes.ANNOTATION_ARRAY_INIT);
```

Comment:
This isn't exactly correct. Java allows multi-value annotation properties to be defined with a single value. Ex, we have this all over the place:

```java
@Metadata(properties = MetadataProperties.FLUENT)
public class TokenRequestContext {
```

Reply:
thanks! fixed in the next line